invenio_rdm_records/services/permissions.py

```diff
@@ -136,6 +137,8 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
     #
     # Allow to remove a community from a record
     can_delete_community = can_manage
+    # Allow to remove records from a community
+    can_delete_record = [CommunityCurators()]
```

Comment:
@kpsherva permission naming might be worth a discussion at some point:
- `can_< action >_<obj>` vs `can_<obj>_< action >`
- more important, base classes are only `can_<action>`, which means that records and communities are `can_read`, `can_delete`, etc. However, then here we need to choose between `can_delete_record` or `can_delete_record_from_community` (which is a super long name). I've no preference, nor magic solution. But it can become messy very fast. In any case, this is separate and should not block this PR.

Reply:
I understand that it is a REST API `DELETE` operation or service `delete` operation, but I would propose to use `can_remove...` instead of delete, to avoid misunderstandings...